# 1.requests 모듈 : HTTP프로토콜을 사용할 수 있게 해주는 모듈
* requsest모듈 사용 설명서
[requests](https://requests.readthedocs.io/en/latest/user/quickstart/)

# 2.requests 모듈 설치
* conda install requests
* pip install requests

# 3. requests 모듈 사용법

* 1) 모듈 불러오기: import requests
* 2) url 변수에 저장
* 3) url 파라미터
  * payload = {파라미터1 : value, 파라미터2 : value}
* 4) headers
  * headers = {key : value}
* 5) HTTP 요청 보내기 get, post
  * get: r = requests.get(url, params=payload, headers=headers)
  * post: r = requests.post(url, data={key:value})
* 6) 만들어진 url 확인: print(r.url)
* 7) 응답 코드 확인: print(r.status_code)
* 8) 응답 요소 출력: r.text, r.content, r.json()

In [ ]:
import requests
url =""
payload = dict()
r = requests.get(url, params=payload)
print(r.url)
print(r.status_code)
response = r.text # r.content, r.json()

# requests 모듈로 1페이지만 가져오기

In [ ]:
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
headers = {"X-Naver-Client-Id": Client_Id,
           "X-Naver-Client-Secret": Client_Secret}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
response = r.json()

In [27]:
start_num2 = 1
for s_num in range(11):
    if start_num2 < 901:
        print(start_num2)
        start_num2 += 100
    elif start_num2 == 901:
        print(start_num2)
        start_num2 += 99
        print(start_num2)

1
101
201
301
401
501
601
701
801
901
1000


In [33]:
import requests
start_num = 1
result_json = []
for page in range(11):
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    result_json.append(response)
    if start_num < 901:
        start_num += 100
    elif start_num == 901:
        start_num += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

* 모든 검색 결과를 가져오기 위해서는 여러 페이지 정보를 수집
* 전체 결과 / 1페이지 표시 개수 + 1 = 전체 페이지수
* start 부분 + 1페이지 표시 개수 = 다음 페이지 데이터 수집 가능
* 반복문을 이용해서 start 에 100씩 더해서 최대 1000이 되도록 하면 최대한 많은 데이터 수집 가능

In [50]:
import requests
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
payload = dict(query="핀테크", display=100, start=1, sort="sim")
headers = {"X-Naver-Client-Id": Client_Id,
           "X-Naver-Client-Secret": Client_Secret}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
response = r.json()


https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200


In [51]:
pages = response['total'] // response['display'] + 1
pages

2

In [52]:
start_num = 1
response_json2 = []
for page in range(pages):
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="핀테크", display=100, start=start_num, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    pages = response['total'] // response['display'] + 1
    response_json2.append(response)
    if start_num == 1000:
        break
    if start_num < 901:
        start_num += 100
    elif start_num == 901:
        start_num += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200


In [54]:
response_json2[1]

{'lastBuildDate': 'Tue, 01 Apr 2025 11:28:31 +0900',
 'total': 106,
 'start': 101,
 'display': 6,
 'items': [{'title': '핀테크 / 블록체인 기술이 적용된 온라인 시스템 전자결제 / 보안 / 인증 시장 전망과 핵심기술 동향',
   'link': 'https://search.shopping.naver.com/book/catalog/32438161048',
   'image': 'https://shopping-phinf.pstatic.net/main_3243816/32438161048.20221227205547.jpg',
   'author': 'TF정보분석실',
   'discount': '342000',
   'publisher': '트렌드포커스',
   'pubdate': '20190429',
   'isbn': '9791188774074',
   'description': '4차 산업혁명 시대에 핀테크와 블록체인이 핵심기술로 떠오르고 있습니다.\n핀테크는 금융서비스 가치와 혁신을 향상시킬 수 있는 비즈니스 모델로 대두되고 있으며 \n블록체인은 세계경제포럼(World Economic Forum, WEF)에서 4차 산업혁명 시대의 핵심\n기술로 선정될 정도로 인정받고 있고 올해까지 전 세계 은행의 80%가 블록체인 기술을 도\n입할 것으로 전망되고 있습니다.\n\n핀테크(fintech)는 금융서비스 도메인에서 IT 기술을 기반으로 새로운 응용프로그램, 프로\n세스, 제품, 비즈니스 모델 또는 새로운 서비스를 제공하는 것으로 이를 통해 금융서비스의 \n가치와 혁신을 향상시키고 있으며 IT기술의 발전에 따라 간편 결제, 대출 프레임워크, 크라\n우드 펀딩등 다양한 형태의 금융서비스가 출시되고 있습니다.\n그리고 블록체인은 금융 분야를 비롯하여 제조 및 유통, 공공서비스, 사회?문화 부문의 \n전영역에 걸쳐 혁신적인 변화를 일으키는 새로운 패러다임이라 말할 수 있습니다.\

# While True로 전체데이터 가져오기

In [56]:
page = 1
start_num2 = 1
response_json3 = []
while True:
    url = "https://openapi.naver.com/v1/search/news.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    payload = dict(query="삼성증권 WM 고객 예탁자산", display=100, start=start_num2, sort="sim")
    headers = {"X-Naver-Client-Id": Client_Id,
               "X-Naver-Client-Secret": Client_Secret}
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    response = r.json()
    total_pages = (response['total'] // 100) + 1
    response_json3.append(response)
    if start_num2 == 1000:
        break
    if start_num2 < 901:
        start_num2 += 100
    elif start_num2 == 901:
        start_num2 += 99
    
    if page < total_pages:
        page += 1
    else:
        break
print(len(response_json3))

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

# 블로그, 책, 뉴스 동시에 수집하기

In [59]:
import time
response_json3 = []
for category in ["blog", "news", "book"]:
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query="핀테크", display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url)
        print(r.status_code)
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        response_json3.append(response)
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.5)
print(len(response_json3))

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&di

# 블로그, 뉴스, 책 데이터 각각 저장하기

In [75]:
import time
keyword = input("검색할 키워드를 입력해 주세요 ")
response_blog = []
response_news = []
response_book = []
for category in ["blog", "news", "book"]:
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query=keyword, display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url, end="\r")
        print(r.status_code, end="\r")
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        
        if category == 'blog':
            response_blog.append(response)
        elif category == "news":
            response_news.append(response)
        elif category == "book":
            response_book.append(response)
                        
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.5)
print(len(response_blog))
print(len(response_news))
print(len(response_book))

검색할 키워드를 입력해 주세요 인공지능
110ps://openapi.naver.com/v1/search/book.json?query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&display=100&start=1000&sort=sim
11
11


In [7]:
def json_unpack(all_data):
    final_result = {}
    for page in all_data:
        for item in page['items']:
            for key, value in item.items():
                final_result.setdefault(key, []).append(value)
    df = pd.DataFrame(final_result)    
    return df

In [78]:
for jsons, cat in zip([response_blog, response_news, response_book],["blog", "news", "book"]):
    unpack_result = json_unpack(jsons)
    unpack_result.to_csv(f"naver_{cat}_result.csv", encoding="utf-8", index=False)

# 3개 카테고리 수집 코드를 함수화하기

In [10]:
import requests
import time
import os
import pandas as pd

In [11]:
def naver_search(keyword, category):
    response_result = []
    page = 1
    start_num2 = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        payload = dict(query=keyword, display=100, start=start_num2, sort="sim")
        headers = {"X-Naver-Client-Id": Client_Id,
                   "X-Naver-Client-Secret": Client_Secret}
        r = requests.get(url, params=payload, headers=headers)
        print(r.url, end="\r")
        print(r.status_code, end="\r")
        response = r.json()
        total_pages = (response['total'] // 100) + 1
        
        response_result.append(response)
                        
        if start_num2 == 1000:
            break
        if start_num2 < 901:
            start_num2 += 100
        elif start_num2 == 901:
            start_num2 += 99

        if page < total_pages:
            page += 1
        else:
            break
        time.sleep(0.2)
    
    return response_result


In [12]:
import time
keyword = input("검색할 키워드를 입력해 주세요 ")
for category in ["blog", "news", "book"]:
    naver_search_result = naver_search(keyword, category)
    result_df = json_unpack(naver_search_result)
    display(result_df)

검색할 키워드를 입력해 주세요 인공지능


title  \
0        제주도 1박2일 여행 스케줄 <b>인공지능</b> 프로그램 챗GPT의 도움을...    
1                 학점은행제 <b>인공지능</b> 학위 취득 직장인도 할 수 있네요!   
2                               <b>인공지능</b>학과 체계적인 대비방법   
3                    고양이홈캠 도기보기 도기캠 <b>인공지능</b>으로 스마트하게   
4        <b>인공지능</b> 스피커 종류 KT 기가지니4 스펙 요금제 알아보기(lg...    
...                                                 ...   
1095                                    <b>인공지능</b>과 사진   
1096                  <b>인공지능</b>을 활용해 아이 맞춤형 영어 교재 만들기   
1097                  관악구,<b>인공지능</b>(AI) 아동 그림 심리검사서비스   
1098  AI <b>인공지능</b> 로봇 알파미니(Alpha Mini)로 배우는 <b>인공지...   
1099  [한국폴리텍대학 성남캠퍼스] &lt;<b>인공지능</b>소프트웨어과&gt; GV 2...   

                                                   link  \
0                    https://jongamk.tistory.com/526077   
1              https://blog.naver.com/2_dl/223813189505   
2     https://blog.naver.com/blackhole_chinese/22378...   
3       https://blog.naver.com/sunlovely09/223817969436   
4         https://blog.naver.com/minsha321/223816637896   
...                                                 ...   
1095      https://blog.naver.com/commbooks/223631394459   
1096   https://blog.naver.com/jonathanshim/223742139185   
1097        https://blog.naver.com/coreap6/223658753251   
1098          https://blog.naver.com/mdhb7/223790924778   
1099  https://blog.naver.com/poly_seongnam/223666247611   

                                            description     bloggername  \
0     <b>인공지능</b> 프로그램 챗GPT로 떠난 제주 1일차 과연 제주도 여행을 챗G...  핑구야 날자의 IT와 일상   
1     등에 인공적으로 구현한, 모방한 기능을 갖춘 컴퓨터 시스템으로 이와 같이 지능을 만...       2_dl님의블로그   
2     <b>인공지능</b>학과 합격자의 후기를 소개합니다 <b>인공지능</b>학과 체계적...       묘한 교육 연구소   
3     <b>인공지능</b> AI 기능도 추가되어 있어서 반려동물을 좀더 세심하게 맞춤형으...         소중한 나날들   
4     &quot; &quot; <b>인공지능</b> 스피커, 어떤 제품을 선택해야 좋을까...         일해라 절해라   
...                                                 ...             ...   
1095  <b>인공지능</b>은 손과 카메라를 넘어 이미지를 생성하며, GAN 같은 기술은 ...        커뮤니케이션북스   
1096  <b>인공지능</b>이 고도로 발전하면서 이제 교육계의 오랜 염원이던 학생 맞춤형 ...     심정섭의 학군과 교육   
1097  관악구,‘<b>인공지능</b>(AI) 아동 그림 심리검사서비스’로 아동학대 예방 총...           새서울신문   
1098  알파미니는 소형 <b>인공지능</b> 로봇으로, 27cm의 작은 크기를 가지고 있어...            모두행복   
1099  저는 <b>인공지능</b>소프트웨어과 홍보를 맡게 된 박서현입니다. 성남시에서 주최...   한국폴리텍대학 성남캠퍼스   

                           bloggerlink  postdate  
0         https://jongamk.tistory.com/  20250323  
1                  blog.naver.com/2_dl  20250328  
2     blog.naver.com/blackhole_chinese  20250306  
3           blog.naver.com/sunlovely09  20250401  
4             blog.naver.com/minsha321  20250331  
...                                ...       ...  
1095          blog.naver.com/commbooks  20241024  
1096       blog.naver.com/jonathanshim  20250130  
1097            blog.naver.com/coreap6  20241113  
1098              blog.naver.com/mdhb7  20250310  
1099      blog.naver.com/poly_seongnam  20241119  

[1100 rows x 6 columns]

title  \
0           &quot;AI 에어컨 수요 적극 대응&quot;…삼성전자, 생산라인 풀가동   
1               애플 AI 한국어 지원 시작했지만…이용자 맞춤형 '시리'는 언제 나와   
2          올트먼 &quot;GPU 녹는다&quot; 행복한 비명…'지브리 프사' 대유행   
3            중앙대 <b>인공지능</b>인문학연구소, '인문한국 3.0 지원사업' 선정   
4              금목걸이 절도범 <b>인공지능</b> 선별 관제 시스템에 '딱 걸렸네'   
...                                                ...   
1095  경륜·경정 누리집(홈페이지) 전면 재구축 ⋯ <b>인공지능</b> 기반 서비스 강...   
1096         KAI, <b>인공지능</b> 데이터솔루션 기업 젠젠AI 지분 10% 확보   
1097            한국표준협회, 고누아이와 '<b>인공지능</b> 인재 양성과정' 개최   
1098      IBK기업은행, '제3회 해군해병대 <b>인공지능</b>(AI) 경진대회' 후원   
1099                    AI스타트업 젠젠에이아이, KAI서 120억 투자 유치   

                                           originallink  \
0            https://www.etoday.co.kr/news/view/2458159   
1     https://www.newsis.com/view/NISX20250401_00031...   
2        https://www.hankyung.com/article/202504012315g   
3     https://edu.donga.com/news/articleView.html?id...   
4            https://www.news1.kr/local/jeonbuk/5738541   
...                                                 ...   
1095              https://www.mk.co.kr/article/11259896   
1096  https://biz.sbs.co.kr/article_hub/20000221496?...   
1097  https://www.enetnews.co.kr/news/articleView.ht...   
1098  https://weekly.cnbnews.com/news/article.html?n...   
1099     https://www.hankyung.com/article/2025030702041   

                                                   link  \
0            https://www.etoday.co.kr/news/view/2458159   
1     https://n.news.naver.com/mnews/article/003/001...   
2     https://n.news.naver.com/mnews/article/015/000...   
3     https://edu.donga.com/news/articleView.html?id...   
4     https://n.news.naver.com/mnews/article/421/000...   
...                                                 ...   
1095  https://m.sports.naver.com/general/article/410...   
1096  https://n.news.naver.com/mnews/article/374/000...   
1097  https://www.enetnews.co.kr/news/articleView.ht...   
1098  https://weekly.cnbnews.com/news/article.html?n...   
1099  https://n.news.naver.com/mnews/article/015/000...   

                                            description  \
0     이투데이=박민웅 기자 | 삼성전자가 올해 무더위에 따른 에어컨 수요에 신속하게 대응...   
1     출시 약 반년 만에 한국어 지원을 시작한 애플의 <b>인공지능</b>(AI) 시스템...   
2     오픈AI가 지난달 25일 선보인 ‘챗GPT-4o’의 새 <b>인공지능</b>(AI)...   
3     중앙대학교 <b>인공지능</b>인문학연구소가 교육부 산하 한국연구재단이 지원하는 '...   
4     중고 거래 앱 '당근마켓'을 통해 금목걸이를 건네받은 뒤 오토바이를 타고 도주한 4...   
...                                                 ...   
1095  이번 경륜·경정 누리집 개편은 <b>인공지능</b>(AI) 기술을 활용한 순위 예측...   
1096  한국항공우주산업(KAI)은 6일 <b>인공지능</b>(AI) 기반 국방 합성 데이터...   
1097  한국표준협회(회장 문동민, 협회)는 인텔 코리아, <b>인공지능</b>(AI) 교육...   
1098  사진=IBK기업은행 IBK기업은행(은행장 김성태)이 지난 7일 해군해병대 AI 역량...   
1099  <b>인공지능</b>(AI) 스타트업 젠젠에이아이가 한국항공우주산업(KAI)으로부터...   

                              pubDate  
0     Wed, 02 Apr 2025 08:32:00 +0900  
1     Wed, 02 Apr 2025 06:01:00 +0900  
2     Tue, 01 Apr 2025 10:30:00 +0900  
3     Tue, 01 Apr 2025 16:54:00 +0900  
4     Tue, 01 Apr 2025 11:23:00 +0900  
...                               ...  
1095  Tue, 11 Mar 2025 08:01:00 +0900  
1096  Fri, 07 Mar 2025 10:58:00 +0900  
1097  Tue, 18 Mar 2025 15:12:00 +0900  
1098  Mon, 10 Mar 2025 14:42:00 +0900  
1099  Fri, 07 Mar 2025 17:32:00 +0900  

[1100 rows x 5 columns]

title  \
0                           인공지능 (수학적 이해에서 알고리즘까지, 개정판)   
1     교실 밖 인공지능 수업 (우리의 상상을 미래의 현실로 만들어가는 인공지능의 세계를 ...   
2                        AI 리터러시 (인공지능 필수 지식부터 완벽 활용까지)   
3         된다! 하루 만에 끝내는 챗GPT 활용법 (인공지능에게 일 시키고 시간 버는 법)   
4                                인공지능 시대의 컴퓨터 개론 (개정8판)   
...                                                 ...   
1095       모두의 인공지능 with 스크래치 (누구나 쉽게 만들면서 배우는 인공지능 기초)   
1096                               브리태니커 만화 백과 51: 인공지능   
1097                           4차 산업혁명과 인공지능 영어교육 (제2판)   
1098                                인공지능과 법률 서비스 분야의 혁신   
1099                       AI의 도전 (인공지능의 위협과 공존 그리고 기회)   

                                                   link  \
0     https://search.shopping.naver.com/book/catalog...   
1     https://search.shopping.naver.com/book/catalog...   
2     https://search.shopping.naver.com/book/catalog...   
3     https://search.shopping.naver.com/book/catalog...   
4     https://search.shopping.naver.com/book/catalog...   
...                                                 ...   
1095  https://search.shopping.naver.com/book/catalog...   
1096  https://search.shopping.naver.com/book/catalog...   
1097  https://search.shopping.naver.com/book/catalog...   
1098  https://search.shopping.naver.com/book/catalog...   
1099  https://search.shopping.naver.com/book/catalog...   

                                                  image      author discount  \
0     https://shopping-phinf.pstatic.net/main_530905...         오상훈    26730   
1     https://shopping-phinf.pstatic.net/main_337008...         김현정    12600   
2     https://shopping-phinf.pstatic.net/main_516051...         김용성    22500   
3     https://shopping-phinf.pstatic.net/main_493627...  프롬프트 크리에이터    15480   
4     https://shopping-phinf.pstatic.net/main_365674...     김대수^김경동    25760   
...                                                 ...         ...      ...   
1095  https://shopping-phinf.pstatic.net/main_324362...         이영호    16200   
1096  https://shopping-phinf.pstatic.net/main_324363...      봄봄 스토리     9900   
1097  https://shopping-phinf.pstatic.net/main_324927...         김인석    22500   
1098  https://shopping-phinf.pstatic.net/main_324776...         양종모    22500   
1099  https://shopping-phinf.pstatic.net/main_324670...         현대원    18000   

     publisher   pubdate           isbn  \
0           홍릉  20250311  9791156002635   
1           궁리  20220725  9788958207788   
2          프리렉  20241202  9788965404026   
3      이지스퍼블리싱  20240729  9791163036234   
4         생능출판  20221227  9788970506173   
...        ...       ...            ...   
1095        길벗  20190918  9791160509052   
1096   미래엔아이세움  20170822  9788937888816   
1097     한국문화사  20210330  9791166850011   
1098    한국학술정보  20210331  9791166033940   
1099      나남출판  20210410  9791197127984   

                                            description  
0                                                        
1     다가올 새로운 미래를 준비하는 청소년들을 위해 \nIT전문가가 현장에서 들려주는 생...  
2     AI 시대에 필수로 알아야 할 인공지능의 원리와 개념부터\n학생·직장인·공무원·교육...  
3     《된다! 하루 만에 끝내는 챗GPT 활용법》이 GPT-4o를 반영한 전면 개정 2판...  
4     제1장에서는 디지털 혁명과 4차 산업혁명의 시대를 다루었다. 인공지능 기술, 디지털...  
...                                                 ...  
1095  개념은 쉽게, 실습은 재밌게! \n남녀노소 누구나 볼 수 있는 실용적인 인공지능 입...  
1096  미국에서 인공지능을 연구하는 공 박사는 휴가를 맞아 쌍둥이 조카 슬기와 로운이의 집...  
1097  이 책은 영어교육을 담당하는 전문가들이 4차 산업혁명 시대 정신에 부합하는 영어교육...  
1098  인공지능의 개념과 개발 동향부터, 법률 서비스 영역에서의 인공지능 알고리즘 활용까지...  
1099  본격화한 인공지능의 도전, 어떻게 대응할 것인가?\n새로운 ‘초지능의 물결’과 인간...  

[1100 rows x 9 columns]

In [ ]:
len(naver_search_result)

In [17]:
import requests
url = "https://openapi.naver.com/v1/search/blog"
payload = {'query': '핀테크', 'display' : 100, 'start': 1, 'sort' : "sim"}
headers = {"X-Naver-Client-Id" :"sk2iKdMANqpvG3Pa2WIJ", "X-Naver-Client-Secret" : "by6xHZcrlB"}
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
data

https://openapi.naver.com/v1/search/blog?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200


{'lastBuildDate': 'Mon, 31 Mar 2025 14:38:57 +0900',
 'total': 456049,
 'start': 1,
 'display': 100,
 'items': [{'title': '<b>핀테크</b> 투자 후기',
   'link': 'https://blog.naver.com/kan1687/223812724931',
   'description': '<b>핀테크</b> 투자, 아직도 망설이고 계신가요?  여러분 혹시 이런 고민 해보신 적 있나요? &quot;<b>핀테크</b> 분야가 떴다는데, 지금 시작해도 괜찮을까? 좀 더 상황을 지켜봐야 하나?&quot; 사실 이런 생각은 투자를 고민하는 우리... ',
   'bloggername': '우정잉♥',
   'bloggerlink': 'blog.naver.com/kan1687',
   'postdate': '20250329'},
  {'title': '[<b>핀테크</b>] 2024년 <b>핀테크</b> 업계 결산: 놀라운 3가지 변화',
   'link': 'https://blog.naver.com/xoom777/223708685044',
   'description': '동안 <b>핀테크</b> 업계에는 예상치 못한 변화의 바람이 불었습니다. 한동안 주춤했던 트렌드가... 올 한 해 <b>핀테크</b> 업계를 뜨겁게 달군 3가지 트렌드에 대해 자세히 살펴보겠습니다. 암호화폐의 귀환, 기관... ',
   'bloggername': 'Ich fange an',
   'bloggerlink': 'blog.naver.com/xoom777',
   'postdate': '20241229'},
  {'title': '[인사이트 캠퍼스] 2025 동계 <b>핀테크</b> 캠프 (with 전주대학교... ',
   'link': 'https://blog.naver.com/fininsight/223734638126',
   'description': '전주대학교 친구들이 직접 여의

In [6]:
payload

{'query': '핀테크', 'display': 100, 'start': 1, 'sort': 'sim'}

In [9]:
headers

{'client_id': 'sk2iKdMANqpvG3Pa2WIJ', 'client_secret': 'by6xHZcrlB'}

네이버 검색 API를 이용해서 책을 검색하고 결과를 데이터프레임으로 만드세요.
* 검색어는 input()을 이용해서 입력받은 값을 이용하세요.

In [1]:
import requests
import pandas as pd

In [4]:
def json_unpack(data):
    result = {}
    for item in data['items']:
        for key, value in item.items():
            result.setdefault(key, []).append(value)
    df = pd.DataFrame(result)
    return df

In [3]:
keyword = input("검색할 키워드를 입력해 주세요: ")

검색할 키워드를 입력해 주세요: MTS


In [4]:
url = "https://openapi.naver.com/v1/search/news.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
payload = dict(query=keyword, display=100, start=1000, sort="sim")
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
print("전체 검색 데이터 수: ", data['total'], "검색할 페이지수: ", (int(data['total'])//100)+1)
result_df = json_unpack(data)
result_df

https://openapi.naver.com/v1/search/news.json?query=MTS&display=100&start=1000&sort=sim
200
전체 검색 데이터 수:  112911 검색할 페이지수:  1130


title  \
0            현대차증권, <b>MTS</b> '내일'에 AI 종목 브리핑 서비스 출시   
1   현대차증권, <b>MTS</b> ‘내일’ AI 종목 브리핑 서비스 출시…미국 주식까...   
2             현대차증권, <b>MTS</b> ‘내일’ AI 종목 브리핑 서비스 출시   
3             현대차증권, <b>MTS</b> '내일' AI 종목 브리핑 서비스 출시   
4          현대차증권, <b>MTS</b> '내일'에 AI 브리핑 도입…美 공시도 제공   
..                                                ...   
95         키움증권, <b>MTS</b> 매수·매도 체결 1시간여 먹통…“현재 정상화”   
96  키움증권 HTS·<b>MTS</b>, 개장 직후 거래장애… &quot;원인 파악 중...   
97                      키움증권, <b>MTS</b>·HTS 주문 체결 지연   
98  [속보] 키움증권, <b>MTS</b> 매수·매도 체결 오류...&quot;원인 파...   
99                     [속보] 키움증권, <b>MTS</b> 주문 체결 오류   

                                         originallink  \
0   http://www.seoulfn.com/news/articleView.html?i...   
1          https://www.etoday.co.kr/news/view/2465419   
2   https://biz.heraldcorp.com/article/10473749?re...   
3   http://www.shinailbo.co.kr/news/articleView.ht...   
4      https://www.hankyung.com/article/202504255336i   
..                                                ...   
95              https://www.etnews.com/20250403000085   
96  http://www.fieldnews.kr/news/articleView.html?...   
97  https://biz.chosun.com/stock/stock_general/202...   
98  https://www.ddaily.co.kr/page/view/20250403093...   
99        https://news.dealsitetv.com/articles/152974   

                                                 link  \
0   http://www.seoulfn.com/news/articleView.html?i...   
1          https://www.etoday.co.kr/news/view/2465419   
2   https://n.news.naver.com/mnews/article/016/000...   
3   http://www.shinailbo.co.kr/news/articleView.ht...   
4   https://n.news.naver.com/mnews/article/015/000...   
..                                                ...   
95  https://n.news.naver.com/mnews/article/030/000...   
96  http://www.fieldnews.kr/news/articleView.html?...   
97  https://n.news.naver.com/mnews/article/366/000...   
98  https://n.news.naver.com/mnews/article/138/000...   
99        https://news.dealsitetv.com/articles/152974   

                                          description  \
0   현대차증권은 <b>MTS</b>(모바일트레이딩시스템) '내일'에 인공지능(AI)을 ...   
1   이투데이=박정호 기자 | 현대차증권은 모바일트레이딩시스템(<b>MTS</b>) ‘내...   
2   현대차증권은 모바일트레이딩시스템(<b>MTS</b>) ‘내일’에 인공지능(AI)을 ...   
3   현대차증권은 현대차증권 <b>MTS</b>(모바일트레이딩시스템) '내일'에 인공지능...   
4   현대차증권은 자사 모바일트레이딩시스템(<b>MTS</b>) ‘내일’에 인공지능(AI...   
..                                                ...   
95  키움증권 모바일트레이딩시스템(<b>MTS</b>)가 장 시작 직후 매매체결이 안 되...   
96  키움증권의 홈트레이딩 시스템(HTS)과 모바일 트레이딩 시스템(<b>MTS</b>)...   
97  키움증권의 홈·모바일 트레이딩 시스템(HTS·<b>MTS</b>)에서 주문 체결이 ...   
98  키움증권 모바일트레이딩시스템(<b>MTS</b>)에서 장 시장 직후 오류가 생긴 것...   
99  키움증권 모바일트레이딩시스템(<b>MTS</b>)에서 장초반 오류가 발생했다. 3일...   

                            pubDate  
0   Fri, 25 Apr 2025 10:18:00 +0900  
1   Fri, 25 Apr 2025 10:20:00 +0900  
2   Fri, 25 Apr 2025 10:15:00 +0900  
3   Fri, 25 Apr 2025 10:12:00 +0900  
4   Fri, 25 Apr 2025 10:07:00 +0900  
..                              ...  
95  Thu, 03 Apr 2025 09:39:00 +0900  
96  Thu, 03 Apr 2025 09:36:00 +0900  
97  Thu, 03 Apr 2025 09:37:00 +0900  
98  Thu, 03 Apr 2025 09:34:00 +0900  
99  Thu, 03 Apr 2025 09:32:00 +0900  

[100 rows x 5 columns]

# 네이버 API로 검색한 결과 최대한 가져오기
* 1회 네이버 서버에 요청
* 응답에서 total을 찾아서 display로 나누어 총 반복할 페이지 산출
* 반복문 작성 및 종료 조건 설정
* start가 1, 101, 201, 301 처럼 변경되어야 함, start의 max값은 1000
* 1페이지 결과를 데이터 프레임으로 만들고 리스트에 append
* 반복이 끝난 후 리스트에 append된 데이터 프레임을 1개로 합침

# for문을 이용해서 전체 페이지 가져오기

In [62]:
keyword = input("검색할 키워드를 입력해 주세요: ")

검색할 키워드를 입력해 주세요: 파이썬


In [76]:
all_data = []

In [77]:
url = "https://openapi.naver.com/v1/search/book.json"
Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
Client_Secret = "by6xHZcrlB"
headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
payload = dict(query=keyword, display=100, start=1, sort="sim")
r = requests.get(url, params=payload, headers=headers)
print(r.url)
print(r.status_code)
data = r.json()
n_data = data['total']
total_pages = (int(data['total'])//100)+1
print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
result_df = json_unpack(data)
all_data.append(result_df)

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  1113 검색할 페이지수:  12


In [88]:
start = 101
for page in range(total_pages):
    print(f"{page+1}/{total_pages} 수집중", end="\r")
    url = "https://openapi.naver.com/v1/search/book.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
    payload = dict(query=keyword, display=100, start=start, sort="sim")
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    data = r.json()
    n_data = data['total']
    total_pages = (int(data['total'])//100)+1
#     print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
    result_df = json_unpack(data)
    all_data.append(result_df)
    if start == 1000:
        break
    elif start < 901:
        start += 100
    elif start == 901:
        start += 99

https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=101&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=201&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=301&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=401&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=501&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=601&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=701&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&display=100&start=801&sort=sim
200
https://openapi.naver.com/v1/search/book.json?query=%ED%8C%8C%EC%9D%B4%EC%8D%AC&

In [89]:
len(all_data)

11

In [91]:
final_result = pd.concat(all_data)
final_result = final_result.reset_index(drop=True)
final_result

title  \
0     혼자 만들면서 공부하는 파이썬 (37개 라이브러리와 API로 보고서, 차트, 게임,...   
1                  혼자 공부하는 파이썬 (1:1 과외하듯 배우는 프로그래밍 자습서)   
2              Do it! 점프 투 파이썬 (중학생도 첫날부터 실습하는 초고속 입문서)   
3        밑바닥부터 시작하는 딥러닝 1(리마스터판) (파이썬으로 익히는 딥러닝 이론과 구현)   
4     50개의 프로젝트로 완성하는 파이썬 업무 자동화 (프로젝트 따라하기로 업무 자동화 ...   
...                                                 ...   
1095  챗GPT와 업무자동화 (1분 만에 끝내는 엑셀, 구글 시트, 파워포인트, 파이썬 자...   
1096  프로그래밍 정석(개념을 잡아주는) (기초 연산부터 고급 문제해결 능력까지 파이썬을 ...   
1097  머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이노...   
1098         파이썬 딥러닝 파이토치(Python Deep Learning PyTorch)   
1099  Let’s Start Coding with Raptor & Python (랩터 & ...   

                                                   link  \
0     https://search.shopping.naver.com/book/catalog...   
1     https://search.shopping.naver.com/book/catalog...   
2     https://search.shopping.naver.com/book/catalog...   
3     https://search.shopping.naver.com/book/catalog...   
4     https://search.shopping.naver.com/book/catalog...   
...                                                 ...   
1095  https://search.shopping.naver.com/book/catalog...   
1096  https://search.shopping.naver.com/book/catalog...   
1097  https://search.shopping.naver.com/book/catalog...   
1098  https://search.shopping.naver.com/book/catalog...   
1099  https://search.shopping.naver.com/book/catalog...   

                                                  image  \
0     https://shopping-phinf.pstatic.net/main_528593...   
1     https://shopping-phinf.pstatic.net/main_325076...   
2     https://shopping-phinf.pstatic.net/main_403540...   
3     https://shopping-phinf.pstatic.net/main_526155...   
4     https://shopping-phinf.pstatic.net/main_537377...   
...                                                 ...   
1095  https://shopping-phinf.pstatic.net/main_386135...   
1096  https://shopping-phinf.pstatic.net/main_324972...   
1097  https://shopping-phinf.pstatic.net/main_324665...   
1098  https://shopping-phinf.pstatic.net/main_324549...   
1099  https://shopping-phinf.pstatic.net/main_324972...   

                          author discount       publisher   pubdate  \
0                            문현일    23400           한빛미디어  20250214   
1                            윤인성    19800           한빛미디어  20220601   
2                            박응용    19800         이지스퍼블리싱  20230615   
3                         사이토 고키    23400           한빛미디어  20250124   
4                           오토코더    19800  위즈앤북(Wiznbook)  20250317   
...                          ...      ...             ...       ...   
1095                         김철수    16200            위키북스  20230321   
1096  제니퍼 캠벨^폴 그리스^제이슨 몬토조^그렉 윌슨        0           에이콘출판  20100504   
1097                         서민우        0             앤써북  20220520   
1098                 이경택^방성수^안상준    14800           정보문화사  20201008   
1099                         김충석        0           앤아이컴즈  20190815   

               isbn                                        description  
0     9791169213271  “파이썬을 배워서 어디에 활용할 수 있을지 궁금했나요?”\n나의 신용카드 내역 분석...  
1     9791162245651  혼자 해도 충분하다! 1:1 과외하듯 배우는 파이썬 프로그래밍 자습서\n\n『혼자 ...  
2     9791163034735  프로그래밍 분야 8년 연속 베스트셀러!\n《Do it! 점프 투 파이썬》 전면 개정...  
3     9791169213387  딥러닝 분야 부동의 베스트셀러!\n머리로 이해하고 손으로 익히는 가장 쉬운 딥러닝 ...  
4     9791198685353  ㆍ퀴즈와 QR 코드 생성부터 나만의 음성 비서 제작까지 다양한 프로젝트 대공개!\n...  
...             ...                                                ...  
1095  9791158394288  코드 한 줄 몰라도 1분 만에 끝내는 엑셀, 구글 스프레드시트, 파워포인트 업무 자...  
1096  9788960771338  『개념을 잡아주는 프로그래밍 정석』은 기본적인 프로그래밍 요소부터 알고리즘, 디버깅...  
1097  9791185553979  〈머신러닝 딥러닝을 이용한 A.I. 자율주행 RC카 프로젝트 with 파이썬+아두이...  
1098  9788956748573  파이썬은 선택이 아닌 필수! 파이토치로 딥러닝 입문하기!\n\n이 책은 딥러닝 기술...  
1099  9788993163568                     ▶ 이 책은 Python(파이썬)을 다룬 이론서입니다.  

[1100 rows x 9 columns]

# While 문으로 데이터 모두 가져오기

In [1]:
import requests
import pandas as pd
from datetime import datetime

In [2]:
keyword = input("검색 키워드를 입력해 주세요. ")

검색 키워드를 입력해 주세요. mpop


In [5]:
all_data2 = []
start = 1

while True:
    url = "https://openapi.naver.com/v1/search/cafearticle.json"
    Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
    Client_Secret = "by6xHZcrlB"
    headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
    payload = dict(query=keyword, display=100, start=start, sort="sim")
    r = requests.get(url, params=payload, headers=headers)
    print(r.url)
    print(r.status_code)
    data = r.json()
    n_data = data['total']
    total_pages = (int(data['total'])//100)+1
    print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
    result_df = json_unpack(data)
    all_data2.append(result_df)
    if start == 1000:
        break
    elif start < 901:
        start += 100
    elif start == 901:
        start += 99
    

https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=101&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=201&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=301&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=401&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=501&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&start=601&sort=sim
200
전체 검색 데이터 수:  2630 검색할 페이지수:  27
https://openapi.naver.com/v1/search/cafearticle.json?query=mpop&display=100&st

In [6]:
len(all_data2)

11

In [7]:
final_result2 = pd.concat(all_data2)
final_result2 = final_result2.reset_index(drop=True)
final_result2

title  \
0                          삼성증권 <b>mpop</b> 어떤거 추천하시나요?   
1                           삼성증권 <b>mpop</b> 쓰시는 분들께 문의   
2                                    삼성증권 <b>mpop</b>에서   
3                                     ㅋㅍㅈ <b>mpop</b>매출   
4                삼성증권 <b>mPop</b>에 <b>mPop</b>-life 기능!   
...                                                 ...   
1095                                     주식 매수 질문(삼성증권)   
1096  [종목뉴스][더퍼블릭][삼성증권] 05월05일 12시06분 어린이날 선물이 고민이라...   
1097  [종목뉴스][청년일보][삼성증권] 05월05일 00시02분 삼성증권, '주식선물하기...   
1098  [종목뉴스][비즈니스포스트][삼성증권] 05월01일 07시03분 [체험기] 삼성증권...   
1099                                             하외주식매도   

                                           link  \
0     http://cafe.naver.com/likeusstock/1128837   
1     http://cafe.naver.com/likeusstock/1156570   
2          http://cafe.naver.com/pointns/516983   
3        http://cafe.naver.com/seacoffee/195585   
4      http://cafe.naver.com/stockschart/978802   
...                                         ...   
1095   http://cafe.naver.com/sharemoney12/12591   
1096  http://cafe.naver.com/bluegrayur0zu/15382   
1097  http://cafe.naver.com/bluegrayur0zu/15180   
1098  http://cafe.naver.com/bluegrayur0zu/13517   
1099   http://cafe.naver.com/sharemoney12/17728   

                                            description  \
0     Ui 좋다는 평이 많아 옮기려고 합니다 그냥 삼성증권도있고 <b>mpop</b>이라...   
1     삼성증권 <b>mpop</b> 차트설정에 갭보정 기능이 들어있을까요? (증권사 어플...   
2     질문드립니다 국내장화면에서는 여기에서 실시간 그래프가 보이는데 해외주식창과 금현물에...   
3     안녕하세요. 오늘 오후 3시이후부터 키오스크 매출이 <b>mpop</b>에 안 잡히...   
4     안녕하세요~ 오랜만에 글을 쓰는데요 오늘은 삼성증권 <b>mPop</b>의 기능 중...   
...                                                 ...   
1095  ㅠㅠ 증거금100%는 변경할 수 없는 걸까요ㅠㅠ 삼성증권<b>mpop</b>을 사용...   
1096  ▣제목 : 어린이날 선물이 고민이라면?‥삼성증권, ‘주식 선물’ 하세요 ▣내용 : ...   
1097  ▣제목 : 삼성증권, '주식선물하기' 서비스 오픈 ▣내용 : 보유 주식 외에도 신규...   
1098  삼성증권의 모바일앱 엠팝(<b>mPOP</b>)을 실행하고 굴링 서비스에 가입하기까...   
1099  삼성증권 <b>mpop</b> 사용중입니다 지식 공유 부탁드려요 아 그리고 여기는 ...   

                     cafename                               cafeurl  
0                  미국 주식이 미래다    https://cafe.naver.com/likeusstock  
1                  미국 주식이 미래다    https://cafe.naver.com/likeusstock  
2     한국/미국주식 : 장기투자를 지향하는...        https://cafe.naver.com/pointns  
3      ★테사모★프랜차이즈카페 사장모임-카...      https://cafe.naver.com/seacoffee  
4     주식차트 연구소(주식 실제계좌 공개/...    https://cafe.naver.com/stockschart  
...                       ...                                   ...  
1095         재테크를 사랑하는 사람들 모임   https://cafe.naver.com/sharemoney12  
1096               모두의 주식 놀이터  https://cafe.naver.com/bluegrayur0zu  
1097               모두의 주식 놀이터  https://cafe.naver.com/bluegrayur0zu  
1098               모두의 주식 놀이터  https://cafe.naver.com/bluegrayur0zu  
1099         재테크를 사랑하는 사람들 모임   https://cafe.naver.com/sharemoney12  

[1100 rows x 5 columns]

In [8]:
final_result2.to_csv("네이버검색_카페.csv", encoding="utf-8-sig", index=False)

# 뉴스, 블로그, 책 한번에 검색해서 저장하기

In [5]:
all_data3 = []

for category in ["blog", "news", "cafearticle"]:
    start = 1
    while True:
        url = f"https://openapi.naver.com/v1/search/{category}.json"
        Client_Id = "sk2iKdMANqpvG3Pa2WIJ"
        Client_Secret = "by6xHZcrlB"
        headers = {"X-Naver-Client-Id" : Client_Id, "X-Naver-Client-Secret" : Client_Secret}
        payload = dict(query=keyword, display=100, start=start, sort="sim")
        r = requests.get(url, params=payload, headers=headers)
        print(r.url)
        print(r.status_code)
        data = r.json()
        n_data = data['total']
        total_pages = (int(data['total'])//100)+1
        print("전체 검색 데이터 수: ", n_data, "검색할 페이지수: ", total_pages )
        result_df = json_unpack(data)
        all_data3.append(result_df)
        if start == 1000:
            break
        elif start < 901:
            start += 100
        elif start == 901:
            start += 99
final_result3 = pd.concat(all_data3)
final_result3 = final_result3.reset_index(drop=True)
final_result3

https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=1&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=101&sort=sim
200
전체 검색 데이터 수:  456149 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=201&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=301&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=401&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=100&start=501&sort=sim
200
전체 검색 데이터 수:  456148 검색할 페이지수:  4562
https://openapi.naver.com/v1/search/blog.json?query=%ED%95%80%ED%85%8C%ED%81%AC&display=10

title  \
0                                      <b>핀테크</b> 투자 후기   
1       [<b>핀테크</b>] 2024년 <b>핀테크</b> 업계 결산: 놀라운 3가지 변화   
2      [인사이트 캠퍼스] 2025 동계 <b>핀테크</b> 캠프 (with 전주대학교...    
3     [AI <b>핀테크</b>] AI와 빅데이터로 금융 시장을 재편하는 <b>핀테크</...   
4          IPAY <b>핀테크</b>관련주 ETF -미국 CBDC 코인 서비스 실험 확대   
...                                                 ...   
2301  신남방 지역 주요국의 핀테크 발전과 협력방안: 인도, 태국, 말레이시아, 베트남, ...   
2302  [CD] 2018 4차산업 기업.스타트업 주소록 - CD-ROM 1장 (가상현실·드...   
2303  4차산업 기업·스타트업 주소록(2023)(CD) (IOT, VR, 가상현실, 블록체...   
2304  4차산업 기업·스타트업 주소록(2024)(CD) (3D프린터, IOT, VR, 가상...   
2305  전국 체험마을 주소록(2024)(CD) (3D프린터, IOT, VR, 가상현실, 드...   

                                                   link  \
0           https://blog.naver.com/kan1687/223812724931   
1           https://blog.naver.com/xoom777/223708685044   
2        https://blog.naver.com/fininsight/223734638126   
3         https://blog.naver.com/jangpung_/223765345858   
4             https://blog.naver.com/jaso_/223736810791   
...                                                 ...   
2301  https://search.shopping.naver.com/book/catalog...   
2302  https://search.shopping.naver.com/book/catalog...   
2303  https://search.shopping.naver.com/book/catalog...   
2304  https://search.shopping.naver.com/book/catalog...   
2305  https://search.shopping.naver.com/book/catalog...   

                                            description         bloggername  \
0     <b>핀테크</b> 투자, 아직도 망설이고 계신가요?  여러분 혹시 이런 고민 해보...                우정잉♥   
1     동안 <b>핀테크</b> 업계에는 예상치 못한 변화의 바람이 불었습니다. 한동안 주...        Ich fange an   
2     전주대학교 친구들이 직접 여의도까지 방문하여 <b>핀테크</b>(FinTech)의 ...  핀인사이트 (Fininsight)   
3     이제 대출도 AI가 하는 시대 출처 : 파가야 테크놀러지스 홈페이지 금융과 기술이 ...          낭만을 꿈꾸는 일상   
4     IPAY <b>핀테크</b>관련주 ETF -미국 CBDC 코인 서비스 실험 확대 I...            미국주식 퍼주마   
...                                                 ...                 ...   
2301  이 책은 대외경제정책연구원의 신남방 지역 주요국의 핀테크 발전과 협력방안을 다룬 정...                 NaN   
2302  2018 4차산업 기업·스타트업 주소록 CD에는 4차 산업기술, 바이오헬스케어, 창...                 NaN   
2303  4차산업 기업·스타트업 주소록을 분야별, 지역별로 18,460건, IT기업 상세정보...                 NaN   
2304  4차산업 기업·스타트업 주소록을 분야별, 지역별로 19,908건, IT기업 상세정보...                 NaN   
2305  전국 체험마을 주소록을 분야별, 지역별로 33,479건, 관광·여가산업 상세정보를 ...                 NaN   

                    bloggerlink  postdate originallink pubDate  \
0        blog.naver.com/kan1687  20250329          NaN     NaN   
1        blog.naver.com/xoom777  20241229          NaN     NaN   
2     blog.naver.com/fininsight  20250122          NaN     NaN   
3      blog.naver.com/jangpung_  20250218          NaN     NaN   
4          blog.naver.com/jaso_  20250130          NaN     NaN   
...                         ...       ...          ...     ...   
2301                        NaN       NaN          NaN     NaN   
2302                        NaN       NaN          NaN     NaN   
2303                        NaN       NaN          NaN     NaN   
2304                        NaN       NaN          NaN     NaN   
2305                        NaN       NaN          NaN     NaN   

                                                  image        author  \
0                                                   NaN           NaN   
1                                                   NaN           NaN   
2                                                   NaN           NaN   
3                                                   NaN           NaN   
4                                                   NaN           NaN   
...                                                 ...           ...   
2301  https://shopping-phinf.pstatic.net/main_332340...   이충열^이종하^이선호   
2302  https://shopping-phinf.pstatic.net/main_324666...      한국콘텐츠미디어   
2303  https://shopping-phinf.pstatic.net/main_400602...  한국콘텐츠미디어 편집부   
2304  https://shopping-phinf.pstatic.net/main_466775...  한국콘텐츠미디어 편집부   
2305  https://shopping-phinf.pstatic.net/main_466775...  한국콘텐츠미디어 편집부   

     discount